llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td

```diff
@@ -160,9 +160,34 @@ def : TargetInfo<"x86_64", [], [], "", "", 1>;
 
 // TODO: The aspects listed for the intel_gpu targets right now are incomplete; 
 // only the fp16/fp64/atomic64 aspects are listed.
-def : TargetInfo<"intel_gpu_cfl", [AspectFp16, AspectFp64, AspectAtomic64], [8, 16, 32]>;
-def : TargetInfo<"intel_gpu_tgllp", [AspectFp16, AspectAtomic64], [8, 16, 32]>;
-def : TargetInfo<"intel_gpu_pvc", [AspectFp16, AspectFp64, AspectAtomic64], [16, 32]>;
+defvar Fp16Fp64Atomic64 = [AspectFp16, AspectFp64, AspectAtomic64];
+defvar Fp16Atomic64 = [AspectFp16, AspectAtomic64];
+defvar Sg8_16_32 = [8, 16, 32];
+defvar Sg16_32 = [16, 32];
+defvar IntelBaseAspects = [AspectExt_intel_esimd];
+class IntelTarget<string Name, list<Aspect> Aspects, list<int> subGroupSizesList>
+: TargetInfo<Name, IntelBaseAspects # Aspects, subGroupSizesList>;
+def : IntelTarget<"intel_gpu_pvc", Fp16Fp64Atomic64, Sg16_32>;
```

Comment:
Why there are no `_pvc_wg` and other newer architectures?

Reply:
I've added some of this missing targets - see [here](https://github.com/intel/llvm/pull/14188/files#diff-5ad5ea4fce345cb46e102a15973f696af01c8492fed8ae15d633a6382dddc6e0R1556-R1564). For these targets, `ocloc` has them under different names for some reason (e.g. `ocloc` would not recognize `-device pvc_vg`), so I had to add some extra handling in the driver. For `intel_gpu_lnl_m` and `intel_gpu_bmg_g21`, `ocloc` does not seem to accept `lnl_m` or `bmg_g21` (or even `20.4.41` or `20.4.1`, the numeric values of those architectures) as values for `-device` yet. It could be that my environment also has an older `ocloc` version, but either way, I think I will address those architectures in a later PR.